### PR TITLE
Issue #5571: Xpath Mapper : UnsupportedOperationException when several queries are combined

### PR DESCRIPTION
**Pull request by timurt**
https://github.com/checkstyle/checkstyle/issues/5571

### CODE REVIEW

**Comment by romani** [comment]
TC failure:
```
 Initialization issues
'this' reference escaped in object construction (Errors) (1)

src/main/java/com/puppycrawl/tools/checkstyle/xpath
 RootNode.java (1)
56: RootNode() Escape of this during object construction
```

**Comment by romani** [comment]
@timurt , to resolve wercker problem, please rebase on latest master code.

ones all CIs green, I am ok to merge.
Thanks a lot for desire to finish this long lasting xpath implementation.

**Comment by romani** [comment]
@MEZk , please review.

File: src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java [hunk 1]
```diff
@@ -49,6 +49,9 @@
     /** The children. */
     private final List<AbstractNode> children = new ArrayList<>();
 
+    /** The {@code TreeInfo} object. */
+    private TreeInfo treeInfo;
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java]: `TreeInfo` could be `final` if constructor was made to accept it as a parameter, then all sub-classes would just have to give it through `super` call.

**Comment by romani** [comment]
@timurt , please continue.

File: src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java (hunk 1 above)
Comment (timurt) [reply src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java]: Done

**Comment by rnveach** [comment]
@MEZk ping

**Comment by romani** [comment]
If no reply from @MEZk  till Sunday, ok merge this PR without him.

**Pull request merged**